### PR TITLE
Ability to change Storage startup timeout via ENV

### PR DESCRIPTION
**Pull request by harduino**
When running in docker-compose mode, if there are a lot of other docker-services, storage sometimes fails to start and throws error "Unable to start Storage". So, default timeout is increased from 60 to 180 seconds.
Also I've added ability to set custom startup timeout via environment variable STORAGE_STARTUP_TIMEOUT_SECOND

### CODE REVIEW

**Comment by harduino** [comment]
Hi, @haikalpribadi

I cherry-picked my commits with the latest master branch

**Comment by haikalpribadi** [comment]
Thanks, @harduino ! 

@lolski please review when you get a chance.

**Comment by lolski** [comment]
Thank you @harduino !

@haikalpribadi I think the convention is to add configuration to `grakn.properties`, rather than environment variables. Therefore, what if we make it a configuration in `grakn.properties` under `daemon` (that's how we refer to the bootup script):

```
# conf/grakn.properties
...
daemon.storage-bootup-timeout=180
```

**Comment by haikalpribadi** [comment]
Sounds good, @lolski 👍

**Comment by SamuelHassine** [comment]
Hello,

This PR seems to be the good solution to narrow the timeout. But as Julien and I requested (https://github.com/graknlabs/grakn/issues/5570#issuecomment-623534292), please adjust it so the `Grakn Server Start... failed` stops the Docker container. For the moment it runs forever.

**Comment by richard-julien** [comment]
Add please make it also configurable through an environment variable, is really simpler than mounting a configuration file just for that.

**Comment by lolski** [comment]
> Add please make it also configurable through an environment variable, is really simpler than mounting a configuration file just for that.

@haikalpribadi right now, `grakn.properties` cannot be overridden with environment variable (unlike in Grabl), and we have no plan for adding that in the near future.

Should we allow overriding the Storage timeout using an environment variable anyway (as a special case) as per @richard-julien 's comment?

**Comment by richard-julien** [comment]
Im always vote for consistency, in that case you should also take care of JAVA_OPTS option :)

But to be pragmatic for user that have this kind of problem (basically docker user) i clearly vote for a quick win with something like that: 
> private static final int STARTUP_TIMEOUT= Integer.parseInt(SystemProperty.STARTUP_TIMEOUT.value() || "300");

And then you can eventually think about push all the config in grakn.properties with the capability to override all the config through environnement variables.

**Comment by haikalpribadi** [comment]
I think it would be better to follow the current behaviour and configure it through `grakn.properties`, and in the future, we introduce a feature to override `grakn.properties` values with an environment variable.

File: daemon/executor/Storage.java [hunk 1]
```diff
@@ -187,15 +187,24 @@ public boolean isRunning() {
...
 
         // Consume configuration from Grakn config file into Cassandra config file
         initialiseConfig();
 
         Future<Executor.Result> result = daemonExecutor.executeAsync(storageCommand(), graknHome.toFile());
 
-        LocalDateTime timeout = LocalDateTime.now().plusSeconds(STORAGE_STARTUP_TIMEOUT_SECOND);
+        // form storage timeout (default or ENV-value)
+        long startupTimeoutCalculated = STORAGE_STARTUP_TIMEOUT_SECOND;
+        String timeoutEnvValue = System.getenv("STORAGE_STARTUP_TIMEOUT_SECOND");
```
Comment (lolski) [inline daemon/executor/Storage.java:200]: 1. We should declare the new config in `grakn.properties`:

```
# conf/grakn.properties
...
daemon.storage-bootup-timeout=180
```

2. We should declare the new config in https://github.com/graknlabs/grakn/blob/master/common/config/ConfigKey.java#L63

```
public static final ConfigKey<String> DAEMON_STORAGE_BOOTUP_TIMEOUT = key("daemon.storage-bootup-timeout", INT);
```

3. Finally, we should read the value from the config:

```
Integer timeoutEnvValue = graknProperties.getProperty(ConfigKey.DAEMON_STORAGE_BOOTUP_TIMEOUT)
```

**Comment by flyingsilverfin** [comment]
@harduino any chance you've got a moment to make the requested changes? :)

**Comment by harduino** [comment]
@flyingsilverfin what changes are you talking about?

**Comment by flyingsilverfin** [comment]
https://github.com/graknlabs/grakn/pull/5710#discussion_r422254149

**Comment by lolski** [comment]
Perhaps Harduino is currently busy.

@richard-julien if you need it urgently in OpenCTI, feel free to submit a PR based on the guideline [here](https://github.com/graknlabs/grakn/pull/5710#discussion_r422254149) :D

**Comment by lolski** [comment]
Closing this PR since in 2.0 this won't be an issue.

Still, thank you for the contribution @harduino :)

**Pull request closed without merging**